src/ethereum/berlin/state.py

```diff
@@ -361,10 +364,12 @@ def account_has_code_or_nonce(state: State, address: Address) -> bool:
 
     Returns
     -------
-    has_code_or_nonce : `bool`
-        True if if an account has non zero nonce or non empty code,
-        False otherwise.
+    has_code_or_nonce_or_storage : `bool`
+        True if if an account has non zero nonce or non empty code or
+        non empty storage, False otherwise.
     """
+    if address in state._storage_tries:
+        return True
```

Comment:
Would it make sense to introduce a `has_storage(state: State, address: Address)` function?

Either way, I think we need to modify the `ethereum_optimized` monkey patches, since that implementation doesn't have `_storage_tries`.

Adding and patching a `has_storage` function (instead of patching `account_has_code_or_nonce_or_storage`) would probably mean less code in `ethereum_optimized`, which is usually good.

Reply:
Have updated to introduced the `has_storage` function. Haven't implemented any updates to the optimized implementation but made a note in the post-prague re-factoring issue #948